extensions/kafka-client/runtime/src/main/java/io/quarkus/kafka/client/runtime/ui/KafkaUiUtils.java

```diff
@@ -0,0 +1,226 @@
+        for (ConsumerGroupDescription cgd : kafkaAdminClient.getConsumerGroups()) {
+
+            var metadata = kafkaAdminClient.listConsumerGroupOffsets(cgd.groupId())
+                    .partitionsToOffsetAndMetadata().get();
+            var members = cgd.members().stream()
+                    .map(member -> new KafkaConsumerGroupMember(
+                            member.consumerId(),
+                            member.clientId(),
+                            member.host(),
+                            getPartitionAssignments(metadata, member)))
+                    .collect(Collectors.toSet());
+
+            res.add(new KafkaConsumerGroup(
+                    cgd.groupId(),
+                    cgd.state().name(),
+                    cgd.coordinator().host(),
+                    cgd.coordinator().id(),
+                    cgd.partitionAssignor(),
+                    getTotalLag(members),
+                    members));
+        }
+        return res;
+    }
+
+    private long getTotalLag(Set<KafkaConsumerGroupMember> members) {
+        return members.stream()
+                .map(KafkaConsumerGroupMember::getPartitions)
+                .flatMap(Collection::stream)
+                .map(KafkaConsumerGroupMemberPartitionAssignment::getLag)
+                .reduce(Long::sum)
+                .orElse(0L);
+    }
+
+    private Set<KafkaConsumerGroupMemberPartitionAssignment> getPartitionAssignments(
+            Map<TopicPartition, OffsetAndMetadata> topicOffsetMap, MemberDescription member) {
+        var topicPartitions = member.assignment().topicPartitions();
+        try (var consumer = createConsumer(topicPartitions, config)) {
+            var endOffsets = consumer.endOffsets(topicPartitions);
+
+            return topicPartitions.stream()
+                    .map(tp -> {
+                        var topicOffset = Optional.ofNullable(topicOffsetMap.get(tp))
+                                .map(OffsetAndMetadata::offset)
+                                .orElse(0L);
+                        return new KafkaConsumerGroupMemberPartitionAssignment(tp.partition(), tp.topic(),
+                                getLag(topicOffset, endOffsets.get(tp)));
+                    })
+                    .collect(Collectors.toSet());
+        }
+    }
+
+    private long getLag(long topicOffset, long endOffset) {
+        return endOffset - topicOffset;
+    }
+
+    public Map<Integer, Long> getOffset(KafkaOffsetRequest request) throws ExecutionException, InterruptedException {
+        return kafkaTopicClient.getPagePartitionOffset(request.getTopicName(), request.getRequestedPartitions(),
+                request.getOrder());
+    }
+
+    public KafkaAclInfo getAclInfo() throws InterruptedException, ExecutionException {
+        var clusterInfo = clusterInfo(kafkaAdminClient.getCluster());
+        var entries = new ArrayList<KafkaAclEntry>();
+        //TODO: fix it after proper error message impl
```

Comment:
Is it normal you have a TODO here? (it might be, just asking)

Reply:
yes
currently we do not have a generic response to transfer error messages to the UI, we just show a pop-up
we plan to fix it later on